server/bundler/remove-parent-modules-monkeypatch.js

```diff
@@ -0,0 +1,116 @@
+// this is really gross.
+//
+// originally from
+/*
+	MIT License http://www.opensource.org/licenses/mit-license.php
+	Author Tobias Koppers @sokra
+*/
+"use strict"; //eslint-disable-line
+
+var forEach = require( 'lodash/forEach' ),
+	filter = require( 'lodash/filter' ),
+	indexOf = require( 'lodash/indexOf' ),
+	pullAll = require( 'lodash/pullAll' ),
+	nanotimer = require( 'nanotimer' );
+
+var RemoveParentModulesPlugin = require( 'webpack/lib/optimize/RemoveParentModulesPlugin' );
+var Chunk = require( 'webpack/lib/Chunk' );
+var Module = require( 'webpack/lib/Module' );
+
+module.exports = RemoveParentModulesPlugin;
+
+function hasModule( chunk, module, checkedChunks ) {
+	if ( ~ indexOf( chunk.modules, module ) ) {
+		return [chunk];
+	}
+	if ( chunk.entry ) {
+		return false;
+	}
+	const uncheckParents = filter( chunk.parents, function( c ) {
+		return ! checkedChunks.has( c );
+	} );
+	return allHaveModule( uncheckParents, module, checkedChunks );
+}
+
+function allHaveModule( someChunks, module, checkedChunks ) {
+	if ( ! checkedChunks ) {
+		checkedChunks = new Set();
+	}
+	let chunks = new Set();
+	for ( let i = 0; i < someChunks.length; i++ ) {
+		checkedChunks.add( someChunks[i] );
+		const subChunks = hasModule( someChunks[i], module, checkedChunks );
+		if ( ! subChunks ) {
+			return false;
+		}
+		addToSet( chunks, subChunks );
+	}
+	return chunks;
+}
+
+function addToSet( set, items ) {
+	forEach( items, function( item ) {
+		set.add( item );
+	} );
+}
+
+Chunk.prototype._calypso_removeModules = function( modules ) {
```

Comment:
this was an attempt to save some time by bulk removing modules from a chunk. It seems to help a bit? Have not gotten a great handle on how much, as it seems to vary wildly.